src/IJ_mv/IJMatrix_parcsr_device.c

```diff
@@ -228,7 +228,7 @@ hypre_IJMatrixAssembleSortAndReduce1(HYPRE_Int  N0, HYPRE_BigInt  *I0, HYPRE_Big
          make_reverse_iterator(thrust::make_zip_iterator(thrust::make_tuple(I0,    J0))),
          make_reverse_iterator(thrust::device_pointer_cast<char>(X0)+N0),
          make_reverse_iterator(thrust::device_pointer_cast<char>(X) +N0),
-         0,
```

Comment:
@pbauman I am not sure if we want `InitialValueType == BigInt`, since the both the input and output are `char *` (the key is `BigInt`. Maybe `char(0)` or actually just `0`?

Reply:
Tested with `char(0)`, which seemed fine.

Reply:
Oops. I believe what you have is correct @liruipeng. However, I think this is worth capturing in a unit test. It appears the `scan_by_key` API behavior changes will land in Thrust 1.12, which neither CUDA nor ROCm are using yet.

Reply:
I should add, my fear of just leaving the raw `0` is that the compiler takes the type as `int` instead of the correct output type.

Reply:
@pbauman Agree, we should use `char(0)`.

Reply:
@pbauman About unit tests, should we add tests for `scan`, `scan_by_key` by some simple example as in https://github.com/NVIDIA/thrust/issues/1379, or unit tests affected hypre routines somehow? Is it enough to just run hypre's whole regression suite after `module load cuda/11.2`? Thanks!

Reply:
If it were me, I would drop in a unit test (and a unit testing framework if you don't have one). The `ij` test failed for me, but very differently for CUDA and HIP. But, the downside, is then you have to go fishing for where the breakage actually occurs, which can take many hours. I suggested a unit test since we know a behavior change is coming. Just my two cents. But if there's not time/resources, the `ij` should show the failure.